### DISCUSSION TITLE
NextJS integration

### DISCUSSION THREAD

**Discussion by SarthakC** [asteasolutions/zod-to-openapi#109] [Ideas]
Next JS integration would be awesome to have. I would be happy to contribute/raise a PR if I can get some cues from the maintainers. Thanks!

**Comment by AGalabov** [comment]
Hey @SarthakC  thanks for the suggestion. The overall idea of `@asteasolutions/zod-to-openapi` has always been what the name suggests - i.e create openapi schema definitions from zod schemas. Basically we are providing a building block/tool. From that point all spin offs of the library can be made as seaparate npm packages. A ~similar (but more broad) suggestion was made in [this discussion](https://github.com/asteasolutions/zod-to-openapi/discussions/91).

Sadly at this moment in time I don't think that we would be capable to start such an initiative but as I also said in the other discussion this is something that we are interested in in general.

If I am to give you cues it would be to think of it as a separate project that utilizes what the `OpenAPIGenerator` and `OpenAPIRegistry` classes provide and not necessarily as something that is an opt-in for this package itself. The code can still be handled in this repository as a monorepo.
